### PR TITLE
add(gh): Add warp_dart_update.yml

### PR DESCRIPTION
**Pull request by pavlzk**
**What this PR does** 📖
Add github workflow to update warp-dart repository on pushes to main branch of Warp

**Which issue(s) this PR fixes** 🔨
WAPR-130

**Special notes for reviewers** 🗒️
For this workflow we also need token to access to wapr-dart repo and branch named update-from-wapr there

**Additional comments** 🎤

### CODE REVIEW

**Comment by dariusc93** [comment]
I did scan over it and I do have a couple of suggestions

**Comment by pavlzk** [comment]
> I did scan over it and I do have a couple of suggestions

i will be glad for your comments :)

File: .github/workflows/warp_dart_update.yml [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+          sudo apt-get install -y llvm-dev cmake libudev-dev
+      - name: Setup rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+          default: true
+          override: true
+      - run: rustup target add aarch64-linux-android armv7-linux-androideabi x86_64-linux-android i686-linux-android
+      - name: Build headers
+        run: cargo build --features "build-header"
```
[the remark this answers is not in the document]
Comment (dariusc93) [reply .github/workflows/warp_dart_update.yml:28]: Hmm if cbindgen is installed by default with one of the rust actions then it wouldnt be needed to install manually.

File: .github/workflows/warp_dart_update.yml [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        working-directory: ./warp
+      - name: Setup the flutter environment.
+        uses: subosito/flutter-action@v2
+        with:
+          flutter-version: '3.0.5' # you can also specify exact version of flutter
```
Comment (dariusc93) [inline .github/workflows/warp_dart_update.yml:33]: @lgmarchi could you confirm if this is the flutter version that should be used?

**Comment by dariusc93** [comment]
Before merging would there be anything else that would be needed to make sure this is operational?

**Comment by pavlzk** [comment]
> Before merging would there be anything else that would be needed to make sure this is operational?
As i described in notes: we also need token to access to wapr-dart repo and branch named update-from-wapr there

**Comment by tooshel** [comment]
image

We have a token now!

**Pull request merged**